pandas/_libs/reduction.pyx

```diff
@@ -543,7 +544,7 @@ def apply_frame_axis0(object frame, object f, object names,
                 # `piece` might not have an index, could be e.g. an int
                 pass
 
-            results.append(piece)
+            results.append(copy(piece))
```

Comment:
can we infer any cases in which this copy is unnecessary?

Reply:
I think so, possibly check whether `piece` has some simple type like `int`, `str` or `bool`?  Seems there should be a better way though.

Reply:
Can we generalize to checking for a scalar object?  I think we have checks for that.

Reply:
@gfyoung Done, that seems to work